packages/@aws-cdk/aws-rds/lib/serverless/serverless.ts

```diff
@@ -0,0 +1,423 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as kms from '@aws-cdk/aws-kms';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { Resource, Construct, Duration, Token, Annotations, RemovalPolicy, CfnDeletionPolicy } from '@aws-cdk/core';
+import { IClusterEngine } from '../cluster-engine';
+import { IServerlessDatabaseCluster, ServerlessDatabaseClusterAttributes } from '../cluster-ref';
+import { DatabaseSecret } from '../database-secret';
+import { Endpoint } from '../endpoint';
+import { IParameterGroup } from '../parameter-group';
+import { Login, RotationMultiUserOptions } from '../props';
+import { CfnDBCluster, CfnDBSubnetGroup } from '../rds.generated';
+
+/**
+ * Properties to configure a Serverless Cluster
+ */
+export interface ServerlessClusterBaseProps {
+
+  /**
+   * What kind of database to start
+   */
+  readonly engine: IClusterEngine;
+
+  /**
+   * Username and password for the administrative user
+   */
+  readonly masterUser: Login;
+
+  /**
+   * An optional identifier for the cluster
+   *
+   * @default - A name is automatically generated.
+   */
+  readonly clusterIdentifier?: string;
+
+  /**
+   * The number of days during which automatic DB snapshots are retained. Set
+   * to zero to disable backups.
+   *
+   * @default Duration.days(1)
+   */
+  readonly backupRetention?: Duration;
+
+  /**
+   * Name of a database which is automatically created inside the cluster
+   *
+   * @default - Database is not created in cluster.
+   */
+  readonly defaultDatabaseName?: string;
+
+  /**
+   * Indicates whether the DB cluster should have deletion protection enabled.
+   *
+   * @default false
+   */
+  readonly deletionProtection?: boolean;
+
+  /**
+   * Whether to enable the HTTP endpoint for an Aurora Serverless database cluster
+    *
+   * @default false
+   */
+  readonly enableHttpEndpoint?: boolean;
+
+  /**
+   * The VPC that this Aurora Serverless cluster has been created in.
+   *
+   * @default - VPC is created with 2 availability zones (AZs)
+   */
+  readonly vpc: ec2.IVpc;
+
+  /**
+   * Where to place the instances within the VPC
+   *
+   * @default - the VPC default strategy if not specified.
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * Scaling configuration
+   *
+   * @default - None
+   */
+  readonly scaling?: ServerlessScalingOptions;
```

Comment:
It is not because all the properties have defaults when you set the `EngineMode` to be `serverless`. Good point on the documentation though. Was somewhat hesitatnt to add it since scaling defaults are different wrt min and max capacity based on engine choice

Reply:
That's fine, let's call out those that are variable as such, and those that are not with their specific values 🙂.